File: writeups/1_Two_Sum/ruby/main.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  h = {}
  nums.each_with_index do |num, i|
    diff = target - num
    if h.has_key?(diff)
      return [h.fetch(diff), i]
    else
      h[num] = i
    end
  end
end

File: writeups/1_Two_Sum/typescript/main.ts
function twoSum(nums: number[], target: number): number[] {
  const h = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const diff: number = target - nums[i];
    if (h.has(diff)) {
      return [h.get(diff), i];
    } else {
      h.set(nums[i], i);
    }
  }
}
